### ISSUE TITLE
Specify the class for detection on pre-trained YOLOv8

### ISSUE THREAD

**Issue by turinaf** [ultralytics/ultralytics#982]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I wanted to use the yolov8 pre-trained model to detect only certain classes, not all the classes on which the model is trained. 
I know we can do this on yolov5 to specify the class. I couldn't do the same with yolov8. 
```
model = torch.hub.load("ultralytics/yolov5", 'custom', path='yolov5s.pt')
model.classes = [0]  # Only person
model.conf = 0.5
```

Is there something similar for YOLOv8?

**Comment by Laughing-q** [comment]
@turinaf 
```python
from ultralytics import YOLO
model = YOLO(f"yolov8n.pt")
model.predict(source="ultralytics/assets/", save=True, classes=[0,1])
```

**Comment by turinaf** [comment]
@Laughing-q hi from Chengdu,
Thank you!

**Comment by kosongdansatu** [comment]
> @turinaf
> 
> ```python
> from ultralytics import YOLO
> model = YOLO(f"yolov8n.pt")
> model.predict(source="ultralytics/assets/", save=True, classes=[0,1])
> ```

where this code replace on yolov8?

**Comment by kosongdansatu** [comment]
I'm Running this code from vscode how to select Specify class?

**Comment by glenn-jocher** [comment]
@kosongdansatu to select specific classes while using YOLOv8 in Visual Studio Code, you can add the "classes" parameter to the `predict` function, and set it to be a list of integers representing the classes you want to detect. For example, if you only want to detect objects from classes 0 and 1, you can set `classes=[0, 1]`.

Here's an example code snippet:

```python
from ultralytics import YOLO

model = YOLO("yolov8.pt")

# Detect objects from classes 0 and 1 only
classes = [0, 1]

# Set the confidence threshold
conf_thresh = 0.5

# Set the source of the input data (e.g., image file, video file, or folder containing images)
source = "path/to/your/data"

# Call the predict function with the specified parameters
results = model.predict(source=source, save=True, classes=classes, conf=conf_thresh)
```

Please note that you need to replace "path/to/your/data" with the actual path to your data. Additionally, you can adjust the value of the "conf_thresh" parameter to modify the sensitivity of the model.

**Comment by kosongdansatu** [comment]
thank you  soo much  glenn

**Comment by glenn-jocher** [comment]
@kosongdansatu you're welcome! If you have any further questions or concerns about YOLOv8, don't hesitate to ask. We're always here to help.

**Comment by Jesly-Joji** [comment]
Can we add the "classes" parameter to the val function?

**Comment by pderrenger** [comment]
Yes, you can add the "classes" parameter to the `val` function in YOLOv8 to validate only specific classes. This allows you to focus on the performance of particular object categories. Here's how you can do it:

```python
from ultralytics import YOLO

model = YOLO("yolov8.pt")

# Validate only specific classes, e.g., classes 0 and 1
results = model.val(data="path/to/your/data.yaml", classes=[0, 1])
```

This will ensure that the validation process considers only the specified classes. If you encounter any issues, make sure you're using the latest version of the package. If the problem persists, feel free to reach out. Happy coding!

**Comment by Jesly-Joji** [comment]
Thanks for the answer. 
I tried the above method, but the problem still persists. In the confusion matrix names of all the 80 labels are still there. I want it show confusion matrix with names of labels of only specified classes.
I even have upgraded the package.
![confusion_matrix](https://github.com/user-attachments/assets/00dc101e-aae3-47dc-a818-7220a078fe0f)

**Comment by pderrenger** [comment]
@Jesly-Joji thanks for the update. It sounds like the confusion matrix is still displaying all labels despite specifying classes. This might be a limitation in the current implementation. I'll bring this to the team's attention for further investigation. In the meantime, you might consider manually filtering the confusion matrix results to focus on your specified classes. If you have any other questions or need further assistance, feel free to ask!

**Comment by pedefe** [comment]
Hello,
I call the tracking as following:
results = _model.track(frame, persist=True, classes = [0,1,2,3,5,7], show_labels = False, show_conf = False, show_boxes = False)
and this doesn't change result : not selected class is find, and unwanted information (label,boxes, confidence) is displayed.
Why ?
best regards.

**Comment by glenn-jocher** [comment]
@Jesly-Joji the `val` function currently doesn't support filtering by specific classes for the confusion matrix. You might need to modify the evaluation script to achieve this. If you have further questions, feel free to ask!

**Comment by Jesly-Joji** [comment]
Thanks for the answer.
I would like to ask is there any other function for creating the confusion matrix or do I have to use scikit-learn for it.

**Comment by glenn-jocher** [comment]
You're welcome! Currently, YOLOv8 doesn't have a built-in function for customizing the confusion matrix, so using scikit-learn is a good option for more control over the output.
